src/storage/src/storage_state.rs

```diff
@@ -272,6 +362,11 @@ impl<'w, A: Allocate> Worker<'w, A> {
         }
 
         if !new_uppers.is_empty() {
+            for (id, upper) in &new_uppers {
+                if let Some(handle) = &self.storage_state.sink_handles.get(id) {
+                    handle.downgrade_since(upper.clone());
```

Comment:
I think there should be 2 comments here:
the first should explain what an `upper` is in the context of a sink, and why we downgrade to it (and not before it, like we do with things that _produce_ collections. The second should explain why we do this here, and don't use `AllowCompaction` to drive this.
The doc comments on `SinkHandle` capture that this will be removed with critical since handles, but I think these 2 comments make sense here